src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -72,8 +63,10 @@ public void setupSchema(List<ColumnMetadata> columns) {
     // precision
     int id = 1;
     for (ColumnMetadata column : columns) {
-      Type type = getColumnParquetType(column, id);
-      parquetTypes.add(type);
+      ParquetTypeGenerator.ParquetTypeInfo typeInfo =
+          ParquetTypeGenerator.generateColumnParquetTypeInfo(column, id);
+      parquetTypes.add(typeInfo.getParquetType());
+      this.metadata.putAll(typeInfo.getMetadata());
```

Comment:
is there any concern about overwriting old entries with same key but different values

Reply:
I don't think so, because the keys are column ids (and column_id+obj_enc for objects), which should be unique.